packages/suite/src/components/wallet/AccountTopPanel/RefreshAccountButton.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { useState } from 'react';
+import styled, { css } from 'styled-components';
+import { Icon, Tooltip } from '@trezor/components';
+import { SPIN } from '@trezor/components/src/config/animations';
+import { useActions } from '@suite-hooks';
+import { Translation } from '@suite-components';
+import { syncAccounts } from '@wallet-actions/blockchainActions';
+import type { Account } from '@wallet-types';
+
+const StyledIcon = styled(Icon)`
+    ${({ useCursorPointer }) =>
+        !useCursorPointer &&
+        css`
+            animation: ${SPIN} 0.5s infinite ease-in-out forwards;
+        `}
+`;
+
+type RefreshAccountButtonProps = {
+    account: Account;
+};
+
+export const RefreshAccountButton = ({ account }: RefreshAccountButtonProps) => {
+    const { sync } = useActions({ sync: syncAccounts });
+
+    const [isUpdating, setIsUpdating] = useState(false);
+
+    const refresh = () => {
+        if (isUpdating) return;
+        setIsUpdating(true);
+        Promise.all([
+            sync(account.symbol),
+            new Promise(resolve => setTimeout(resolve, 1000)),
```

Comment:
what is the purpose of this timeout?

Reply:
Only visual stuff (refresh button animation). It still could change based on designers input.